Swiftz/Reader.swift

```diff
@@ -0,0 +1,149 @@
+//
+//  Reader.swift
+//  Swiftz
+//
+//  Created by Matthew Purland on 11/25/15.
+//  Copyright © 2015 TypeLift. All rights reserved.
+//
+
+import Foundation
+
+/// A type for a `Reader` monad
+public protocol ReaderType {
```

Comment:
I think in many cases protocols are the way to go, but may not be needed in some places.